metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/DbRuleSource.java

```diff
@@ -0,0 +1,161 @@
+  }
+
+  private List<Document> getForTenant(String tenantId) {
+    RequestContext context = RequestContext.forTenantId(tenantId);
+    List<NotificationRule> notificationRules = getNotificationRules(context);
+    Map<String, JsonNode> idMetricAnomalyJsonMap = getMetricAnomalyJsonMap(context);
+    return mergeNotificationRuleEventCondition(notificationRules, idMetricAnomalyJsonMap);
+  }
+
+  private List<NotificationRule> getNotificationRules(RequestContext requestContext) {
+    return requestContext
+        .call(
+            () ->
+                notificationRuleStub.getAllNotificationRules(
+                    GetAllNotificationRulesRequest.newBuilder().build()))
+        .getNotificationRulesList();
+  }
+
+  private Map<String, JsonNode> getMetricAnomalyJsonMap(RequestContext requestContext) {
+    return requestContext
+        .call(
+            () ->
+                this.configServiceBlockingStub.getAllConfigs(
+                    GetAllConfigsRequest.newBuilder()
+                        .setResourceName(this.resourceName)
+                        .setResourceNamespace(this.resourceNamespace)
+                        .build()))
+        .getContextSpecificConfigsList()
+        .stream()
+        .map(ContextSpecificConfig::getConfig)
+        .map(this::convert)
+        .filter(v -> v.getRight() != null)
+        .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+  }
+
+  private List<Document> mergeNotificationRuleEventCondition(
+      List<NotificationRule> notificationRules, Map<String, JsonNode> idMetricAnomalyJsonMap) {
+    return notificationRules.stream()
+        .filter(
+            rule ->
+                idMetricAnomalyJsonMap.containsKey(
+                    rule.getNotificationRuleMutableData().getEventConditionId()))
+        .map(
+            rule -> {
+              try {
+                JsonNode metricAnomalyNode =
+                    idMetricAnomalyJsonMap.get(
+                        rule.getNotificationRuleMutableData().getEventConditionId());
+                JsonNode ruleNode =
+                    OBJECT_MAPPER
+                        .readTree(JsonFormat.printer().print(rule))
+                        .get(NOTIFICATION_RULE_MUTABLE_DATA_KEY);
+                ((ObjectNode) ruleNode).set(AlertTaskConverter.EVENT_CONDITION, metricAnomalyNode);
+                return new JSONDocument(ruleNode);
+              } catch (IOException e) {
+                throw new UncheckedIOException(e);
+              }
+            })
+        .collect(Collectors.toList());
+  }
+
+  private Pair<String, JsonNode> convert(Value value) {
+    JsonNode jsonNode;
+    try {
+      jsonNode = OBJECT_MAPPER.readTree(JsonFormat.printer().print(value));
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+    if (jsonNode.get(ID) == null) {
+      throw new RuntimeException(String.format("id is missing in the object %s", value));
```

Comment:
What will happen on exception? Will this event be skipped or entire operation will be skipped?

Reply:
I think, we should just skip those invalid event config.